### PR TITLE
ipv6 test is enabled in network_test.py [Waiting for util]

### PR DESCRIPTION
**Pull request by vaishnavibhat**
adding function test_ping6 for testing ipv6 ping

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
Depends on https://github.com/avocado-framework/avocado/pull/3569 for this function to work.

**Comment by vaishnavibhat** [comment]
avocado run network_test.py:NetworkTest.test_ping6 -m network_test.yaml
/usr/lib/python3.6/site-packages/avocado_framework-75.1-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : e0070c9eca225cb1e596a564d73ff1111b3639de
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2020-02-18T16.02-e0070c9/job.log
 (1/1) network_test.py:NetworkTest.test_ping6;1500-53e2: PASS (53.08 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 61.78 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2020-02-18T16.02-e0070c9/results.html
[test_ping6.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4218760/test_ping6.log)

**Comment by vaishnavibhat** [comment]
[ipv6.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4455218/ipv6.log)

#avocado run network_test.py:NetworkTest.test_ping6 -m network_test.yaml 
/usr/local/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : b82f563f585463b283ac30c971b429a67dd71a4d
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2020-04-09T14.34-b82f563/job.log
 (1/9) network_test.py:NetworkTest.test_ping6;1500-7e04: PASS (15.45 s)
 (2/9) network_test.py:NetworkTest.test_ping6;2000-b3bd: PASS (30.49 s)
 (3/9) network_test.py:NetworkTest.test_ping6;3000-80a8: PASS (31.07 s)
 (4/9) network_test.py:NetworkTest.test_ping6;4000-2560: PASS (31.44 s)
 (5/9) network_test.py:NetworkTest.test_ping6;5000-106a: PASS (30.86 s)
 (6/9) network_test.py:NetworkTest.test_ping6;6000-7dc6: PASS (30.21 s)
 (7/9) network_test.py:NetworkTest.test_ping6;7000-d0f2: PASS (30.64 s)
 (8/9) network_test.py:NetworkTest.test_ping6;8000-3a61: PASS (38.55 s)
 (9/9) network_test.py:NetworkTest.test_ping6;9000-5f20: PASS (31.07 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 270.17 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2020-04-09T14.34-b82f563/results.html

**Comment by vaishnavibhat** [comment]
Resending the patch with the existing utility (configure_network.py) . 
ping of ipv6 is added as a function in this file. - test_ping6()
No utility change for configure_network needs to be done here.

File: io/net/network_test.py [hunk 1]
```diff
@@ -274,6 +275,27 @@ def test_promisc(self):
...
+            if result.exit_status == 0:
+                result_ip = result.stdout_text
+                try:
+                    for line in result_ip.splitlines():
+                        if "inet6" in line:
+                            ipv6 = line.split()[1].split('/')[0]
+                except UnboundLocalError:
+                    if ipv6 == "":
+                        self.error("unable to get IPV6 of peer")
+                        return False
```
Comment (narasimhan-v) [inline io/net/network_test.py:295]: @vaishnavibhat Would be better if you write a function for finding peer ipv6, similar to `get_peer_interface()` in configure_network,py, and then use that to get peer ipv6 here.

Comment (vaishnavibhat) [reply io/net/network_test.py:295]: > @vaishnavibhat Would be better if you write a function for finding peer ipv6, similar to `get_peer_interface()` in configure_network,py, and then use that to get peer ipv6 here.

Hi Narasimhan, 

I have opened a PR for the same but got some comments asking to use existing functions. Moreover I there is a comment saying configure_network() will get deprecated. So I have use made changes to network_test.py to test ipv6.

**Comment by vaishnavibhat** [comment]
Hi Narasimhan, 
I am stuck at resolving the travis error. Looks like it is complaining about 

License check: disabled
************* Module network_test
E0602:302,15: NetworkTest.test_ping6: Undefined variable 'configure_network'

Can you please help here. 

Thank you.

File: io/net/network_test.py (hunk 1 above)
Comment (narasimhan-v) [reply io/net/network_test.py:295]: Sure then

**Comment by PraveenPenguin** [comment]
@vaishnavibhat @narasimhan-v  why not we use the latest API network.interfaces.ping_check as this API (configure_network) we deprecated and even we adopted in all other IO test cases or in process of adopting

**Comment by narasimhan-v** [comment]
@vaishnavibhat Is this still being pursued via https://github.com/avocado-framework/avocado/pull/3690 or can it be reviewed ?

Please paste job.log with latest code.

**Comment by narasimhan-v** [comment]
@vaishnavibhat I am closing this. Feel free to re-open if needed

**Pull request closed without merging**